src/main/java/de/vette/idea/neos/util/ComposerUtil.java

```diff
@@ -88,4 +85,101 @@ public static Map<String, String> getNamespaceMappings(JsonFile composerFile)
 
         return mappings;
     }
+
+    /**
+     * As defined in Neos\Flow\Package\PackageManager::getPackageKeyFromManifest
+     */
+    @Nullable
+    public static String getPackageKey(JsonFile composerFile)
+    {
+        JsonObject object = (JsonObject) composerFile.getTopLevelValue();
+        if (object == null) {
+            return null;
+        }
+
+        // test extra.neos.package-key
+        var packageKey = getFromPath(object, "extra", "neos", "package-key");
+        if (packageKey instanceof JsonStringLiteral) {
+            if (isPackageKeyValid(((JsonStringLiteral) packageKey).getValue())) {
+                return ((JsonStringLiteral) packageKey).getValue();
+            }
+        }
+
+        // get from path
+        var packageType = getFromPath(object, "type");
+        if (packageType instanceof JsonStringLiteral && isFlowPackageType(((JsonStringLiteral) packageType).getValue())) {
+            var packageKeyFromDirectory = composerFile.getContainingDirectory().getName();
+            if (isPackageKeyValid(packageKeyFromDirectory)) {
+                return packageKeyFromDirectory;
+            }
+        }
+
+        // test autoload configuration
+        var namespaceMapping = getNamespaceMappings(composerFile);
+        if (!namespaceMapping.isEmpty()) {
+            var firstNamespacePath = namespaceMapping.keySet().stream().findFirst();
+            var packageKeyFromNamespace = firstNamespacePath.get()
```

Comment:
It should be checked with firstNamespacePath.isPresent() if it has a value, before accessing

Reply:
You mean instead of checking whether the Map is empty? Is there a java reasoning about one being better? I'm not fluent with java..
Doing both showed a notice that this would be unnecessary.

Reply:
namespaceMapping.isEmpty() should be enough I guess. Only paid attention to the notice for the get() call.